build.sh

```diff
@@ -32,92 +34,113 @@ CAKE_ARGUMENTS=()
 
 # Parse arguments.
 for i in "$@"; do
-    case $1 in
-        -s|--script) SCRIPT="$2"; shift ;;
-        --) shift; CAKE_ARGUMENTS+=("$@"); break ;;
-        *) CAKE_ARGUMENTS+=("$1") ;;
+    case $(echo "$1" | tr '[:upper:]' '[:lower:]') in
+    -s | --script)
+        SCRIPT="$2"
+        shift
+        ;;
+    --usenetcore)
+        echo "Enabling .NET Core build"
+        USE_NETCORE=1
+        shift
+        ;;
+    --)
+        shift
+        CAKE_ARGUMENTS+=("$@")
+        break
+        ;;
+    *)
+        CAKE_ARGUMENTS+=("$1")
+        echo "Using argument $1"
+        ;;
     esac
     shift
 done
 
-# Make sure the tools folder exist.
-if [ ! -d "$TOOLS_DIR" ]; then
-  mkdir "$TOOLS_DIR"
-fi
+if [ "$USE_NETCORE" -eq "1" ]; then
+    dotnet tool restore
+    MONO_EXE=dotnet
+    CAKE_EXE=cake
+else
+    # Make sure the tools folder exist.
+    if [ ! -d "$TOOLS_DIR" ]; then
+        mkdir "$TOOLS_DIR"
+    fi
 
-# Make sure that packages.config exist.
-if [ ! -f "$TOOLS_DIR/packages.config" ]; then
-    echo "Downloading packages.config..."
-    curl -Lsfo "$TOOLS_DIR/packages.config" https://cakebuild.net/download/bootstrapper/packages
-    if [ $? -ne 0 ]; then
-        echo "An error occurred while downloading packages.config."
-        exit 1
+    # Make sure that packages.config exist.
+    if [ ! -f "$TOOLS_DIR/packages.config" ]; then
+        echo "Downloading packages.config..."
+        curl -Lsfo "$TOOLS_DIR/packages.config" https://cakebuild.net/download/bootstrapper/packages
+        if [ $? -ne 0 ]; then
+            echo "An error occurred while downloading packages.config."
+            exit 1
+        fi
     fi
-fi
 
-# Download NuGet if it does not exist.
-if [ ! -f "$NUGET_EXE" ]; then
-    echo "Downloading NuGet..."
-    curl -Lsfo "$NUGET_EXE" https://dist.nuget.org/win-x86-commandline/latest/nuget.exe
+    # Download NuGet if it does not exist.
+    if [ ! -f "$NUGET_EXE" ]; then
+        echo "Downloading NuGet..."
+        curl -Lsfo "$NUGET_EXE" https://dist.nuget.org/win-x86-commandline/latest/nuget.exe
+        if [ $? -ne 0 ]; then
+            echo "An error occurred while downloading nuget.exe."
+            exit 1
+        fi
+    fi
+
+    # Restore tools from NuGet.
+    pushd "$TOOLS_DIR" >/dev/null
+    if [ ! -f "$PACKAGES_CONFIG_MD5" ] || [ "$(cat "$PACKAGES_CONFIG_MD5" | sed 's/\r$//')" != "$($MD5_EXE "$PACKAGES_CONFIG" | awk '{ print $1 }')" ]; then
+        find . -type d ! -name . ! -name 'Cake.Bakery' | xargs rm -rf
+    fi
+
+    . $MONO_EXE "$NUGET_EXE" install -ExcludeVersion
```

Comment:
Forget my previous comment, it seems to be that something is wrong with this call on OSX.
I tested it on Linux (arch linux), but there probably are some differences that I am not aware of between OSX and Linux in the way binaries can be call from a shell script

Reply:
Happy to try something out if you have some ideas.

Reply:
Unfortunately, right now I got no clue at all what to do about it. 😢

Reply:
Ha ha :joy: Not a problem.  I have even less of a clue, since I rely heavily on @devlead when it comes to .sh scripts 😄

Reply:
@gep13 can you give my latest changes a try.
I made some minor changes to the shell script that I believe should change the issue.